modules/internal/ChapelArray.chpl

```diff
@@ -3665,14 +3669,30 @@ module ChapelArray {
 
   // see comment on chpl__unalias for domains
   pragma "unalias fn"
-  inline proc chpl__unalias(ref x: []) {
+  inline proc chpl__unalias(x: []) {
     const isalias = (x._unowned) || (x._value._arrAlias != nil);
 
     if isalias {
-      chpl_replaceWithDeepCopy(x);
+      // Intended to call chpl__initCopy
+      pragma "no auto destroy" var ret = x;
+      chpl__autoDestroy(x);
+      return ret;
+    } else {
+      // Just return a bit-copy/shallow-copy of 'x'
+      pragma "no copy" var ret = x;
+      return ret;
     }
   }
 
+  pragma "unalias fn"
+  inline proc chpl__unalias(x: [])
+  where x._value.isSliceArrayView() {
+    // Intended to call chpl__initCopy
+    pragma "no auto destroy" var ret = x;
+    chpl__autoDestroy(x);
```

Comment:
I'd probably put a comment here saying that since unalias replaces initCopy(auto/initCopy()) the inner value needs to be auto-destroyed. The autoDestroy could be added here or in callDestructors. I wonder if adjusting callDestructors would be better.